lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -669,12 +671,34 @@ private void handleStateRequest() {
 		requestedState = null;
 	}
 
+	private boolean checkConnectionStale() {
+	    if(lastActivity == 0) {
+	        return false;
+        }
+		long now = System.currentTimeMillis();
+		long intervalSinceLastActivity = now - lastActivity;
+		if(intervalSinceLastActivity > (maxIdleInterval + connectionStateTtl)) {
+			/* RTN15g1, RTN15g2 Force a new connection if the previous one is stale */
+			if(connection.id != null) {
+				connection.id = null;
+				connection.key = null;
+				connection.recoveryKey = null;
+			}
+			return true;
+		}
+		return false;
+	}
+
 	private void handleStateChange(StateIndication stateChange) {
 		/* if we have had a disconnected state indication
 		 * from the transport then we have to decide whether
 		 * to transition to closed, disconnected to suspended depending
 		 * on when we last had a successful connection */
 		if(stateChange.state == ConnectionState.disconnected) {
+			if(checkConnectionStale()) {
+				requestState(ConnectionState.suspended);
```

Comment:
Should we not add a `Log.v(TAG)` statement here for debugging purposes?

Reply:
It would be useful in future to know why a connection moved suddenly to suspended

Reply:
@funkyboy FYI @paddybyers explained to me that this on state change handler is for the transport, not the connection, so my comments above are incorrect.  Sorry, I misunderstood